dew/commands/update.py

```diff
@@ -46,28 +47,42 @@ def set_properties_from_args(self, args: ArgumentData, properties: ProjectProper
             properties.build_type = args.build_type
 
     def execute(self, args: ArgumentData, data: CommandData) -> int:
-        dewfile = data.project_parser.parse()
-        depstates = DependencyStateController(data.storage)
-        self.depstates = depstates
-        depstates.load()
+        lock_path = data.storage.join_storage_dir_path('lock')
+        lock = fasteners.InterProcessLock(lock_path)
 
-        is_dirty = data.properties.dirty
-        if is_dirty:
-            data.view.info('Dew project properties have changed. Clearing dependency state information.')
-            depstates.clear()
-            depstates.save()
+        lock_acquired = lock.acquire(blocking=False)
+        try:
```

Comment:
nitpick: Do fastener locks support being used with `with`?
If so, the following would be more pythonic:
```python
with fasteners.InterProcessLock(lock_path) as lock:
    lock_acquired = lock.acquire(blocking=False)
    if lock_acquired:
        ...
```